### DISCUSSION TITLE
Vertical slides with level 2: how to keep showing level 1 heading? (reveal.js via pandoc)

### DISCUSSION THREAD

**Discussion by ppenguin** [hakimel/reveal.js#3684] [Q&A]
I couldn't find out how to get vertical slides (for level 2 headings) while keeping the level 1 heading statically shown, even though IMO it should be the default (or at least easy to achieve).

I'm using `pandoc` with my slides as Markdown, with a slightly customised pandoc template, and a much more customised `css` to get rid of a lot of the default centering stuff.

Also I'd need a way to *not* show an empty slide with every new level 1 heading, i.e. I want to immediately show a level2 slide (as the first vertical slide of a vertical stack) for every new stack.

**Comment by josephdpurcell** [comment]
I wanted to have a markdown file like:

```md
# Title

## Title of Slide 1

Content

## Title of Slide 2

### Vertical slide
```

And then use it in the slides like:

```html
    <div class="reveal">
      <div class="slides">
        <section
          data-markdown="content.md"
          data-separator-vertical="^### "
          data-separator="^## "
        ></section>
      </div>
    </div>
```

And I ran into two problems:

1. The title slides didn't show as titles because the `##` and `###` would get stripped
2. There would be empty slides

So I made two changes to the markdown.js plugin:

1. I preserved the matched content instead of stripping it https://github.com/josephdpurcell/how-to-remove-invasive-plants/commit/078449df4561b5e67e0bae44767b7eafea69f151#diff-af7daa272ca8db800c7d18cde884d64216402e449fe073cc81513881ff1323daR2063
2. I skipped slides with empty content https://github.com/josephdpurcell/how-to-remove-invasive-plants/commit/078449df4561b5e67e0bae44767b7eafea69f151#diff-af7daa272ca8db800c7d18cde884d64216402e449fe073cc81513881ff1323daR2051

And it seems to work: https://josephdpurcell.github.io/how-to-remove-invasive-plants/presentations/2025-01-27-IGC/index.html

Why did I want to do this? Well, if I'm going to use Markdown for my slides I want to write the content like I would in Markdown, which would be to have a hierarchy of Title then a few main points as H2s and underneath each main point some subpoints as H3s. And, to me that maps to the title slide being the H1 and any content before the first H2, then each H2 is a horizontal slide and any H3 is a vertical slide beneath their corresponding H2.

I'm curious why this isn't supported out of the box. I didn't like the idea of having to clutter the Markdown with special demarcations for horizontal and vertical slides. That's more thoughtwork, not less.
